### ISSUE TITLE
Unable to restore wallet in `ord 0.16.0-rc1`

### ISSUE THREAD

**Issue by so7ow** [ordinals/ord#3225]
I am unable to restore a known-good ord wallet using the 0.16.0-rc1 version of ord.

```
 $ ./ord --version
ord 0.16.0-rc1
```
I start a server:
```
 $ ./ord server --http-port 8000
Listening on http://0.0.0.0:8000
```
I leave that running and execute this in another window:

```
 $ ./ord wallet --name TEST123 --server-url http://127.0.0.1:8000 restore --from mnemonic
```
...and then I get a cursor on the new line. I assume I am supposed to type in my space-separated mnemonic here, on one line? When I do, and press enter, the cursor advances to the next line and then nothing happens. Am I doing something wrong, or is this broken?

Restoring the same wallet in ord 0.15.0 works fine:
```
 $ ./ord --version
ord 0.15.0

 $ ./ord wallet --name TEST1234 restore "phrase redacted phrase redacted phrase redacted phrase redacted phrase redacted phrase redacted"
{}
```

**Comment by so7ow** [comment]
I also just tried entering the mnemonic one word per line. That didn't work either.

**Comment by bingryan** [comment]
https://github.com/ordinals/ord/blob/0529f3e4f4715270006c7b03d232ab0cea749970/src/subcommand/wallet/restore.rs#L30

for `read_to_string `, when you input end In the terminal, you can input EOF (End of File) by pressing `Ctrl + D` (on Unix/Linux systems) or `Ctrl + Z` (on Windows systems) .

**Comment by bingryan** [comment]
@so7ow for your example:

```
 $ ./ord wallet --name TEST123 --server-url http://127.0.0.1:8000 restore --from mnemonic
```
after  input your mnemonic, you must input EOF (End of File) by pressing Ctrl + D (on Unix/Linux systems) or Ctrl + Z (on Windows systems).

**Comment by bingryan** [comment]
@raphjaph BTW,  Is `io::stdin(). read_line(&mut buffer)?` a better choice instead of `io::stdin().read_to_string(&mut buffer)?;`

**Comment by so7ow** [comment]
Thank you, I'll give that a try. If that's truly required, some notice to the user would seem to be in order!

**Comment by so7ow** [comment]
> @so7ow for your example:
> 
> ```
>  $ ./ord wallet --name TEST123 --server-url http://127.0.0.1:8000 restore --from mnemonic
> ```
> 
> after input your mnemonic, you must input EOF (End of File) by pressing Ctrl + D (on Unix/Linux systems) or Ctrl + Z (on Windows systems).

That worked! 👍🏻

**Comment by raphjaph** [comment]
I've fixed this and written some documentation in #3237
